src/scenario-context/useScenario.tsx

```diff
@@ -0,0 +1,25 @@
+import { saveScenario } from "../database";
+import { Scenario } from "../page-multi-facility/types";
+import {
+  ScenarioUpdate,
+  useScenarioDispatch,
+  useScenarioState,
+} from "./ScenarioContext";
+
+export default function useScenario() {
+  const dispatch = useScenarioDispatch();
+  const scenario = useScenarioState();
+
+  function updateScenario(scenario: Scenario) {
+    saveScenario(scenario);
```

Comment:
I tried to be clever here and tie in our persistence changes with our state changes.  It didn't really go according to plan and you'll notice that I have to unwind this in the final clean up commit.